pkg/controller/s3/bucket/taggingConfig.go

```diff
@@ -110,3 +118,35 @@ func (in *TaggingConfigurationClient) Delete(ctx context.Context, bucket *v1beta
 	).Send(ctx)
 	return awsclient.Wrap(err, taggingDeleteFailed)
 }
+
+// LateInitialize does nothing because the resource might have been deleted by
+// the user.
+func (in *TaggingConfigurationClient) LateInitialize(ctx context.Context, bucket *v1beta1.Bucket) error {
+	external, err := in.client.GetBucketTaggingRequest(&awss3.GetBucketTaggingInput{Bucket: awsclient.String(meta.GetExternalName(bucket))}).Send(ctx)
+	if err != nil {
+		return awsclient.Wrap(resource.Ignore(s3.TaggingNotFound, err), taggingGetFailed)
+	}
+
+	in.logger.Debug(fmt.Sprintf("called LateInitialize for %s", reflect.TypeOf(in).Elem().Name()))
+
+	// We need the second check here because by default the tags are not set
+	if external.GetBucketTaggingOutput == nil || len(external.TagSet) == 0 {
+		return nil
+	}
+
+	fp := &bucket.Spec.ForProvider
+	if fp.BucketTagging == nil || len(fp.BucketTagging.TagSet) != 0 {
+		fp.BucketTagging = &v1beta1.Tagging{}
+	} else {
+		return nil
+	}
+
+	fp.BucketTagging = GenerateLocalTagging(external.TagSet)
```

Comment:
We can do this only if `len(fp.BucketTagging.TagSet) == 0`

Reply:
In general, it would be better to check if a list is `nil` as opposed to `len() == 0`, since in some cases the user might intend the value to be an empty list, right?